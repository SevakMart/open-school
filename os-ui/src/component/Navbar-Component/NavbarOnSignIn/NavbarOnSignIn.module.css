.mainContent {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  padding: 28px 88px;
  box-sizing: border-box;
  background-color: #F9F9FD;
}
.mainContent h2 {
  font-family: "Helvetica Neue";
  font-style: italic;
  font-weight: 700;
  font-size: 20px;
  line-height: 24px;
  color: #222427;
  margin: 0;
  cursor: pointer;
}
.mainContent .navMainContent {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
}
.mainContent .navMainContent p {
  font-family: "Montserrat";
  font-style: normal;
  font-weight: 500;
  font-size: 16px;
  line-height: 24px;
  color: #5E617B;
  cursor: pointer;
  box-sizing: border-box;
  margin: 0 0 0 32px;
}
.mainContent .navMainContent .bookmarkIconStyle {
  margin: 0 0 0 25px;
  width: 20px;
  height: 20px;
  text-align: center;
}
.mainContent .navMainContent .notificationIcon {
  margin: 0 0 0 25px;
  cursor: pointer;
}
.mainContent .navMainContent .userInfoContent {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  box-sizing: border-box;
  margin: 0 0 0 25px;
}
.mainContent .navMainContent .userInfoContent .avatarLogo {
  width: 32px;
  height: 32px;
  border-radius: 50%;
}
.mainContent .navMainContent .userInfoContent .arrowDownIcon {
  cursor: pointer;
}/*# sourceMappingURL=NavbarOnSignIn.module.css.map */