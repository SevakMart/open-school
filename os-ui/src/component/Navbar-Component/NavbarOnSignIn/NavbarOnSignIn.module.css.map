{"version":3,"sources":["NavbarOnSignIn.module.scss","../../../styles/Mixins/FlexMixins.scss","NavbarOnSignIn.module.css","../../../styles/Mixins/FontMixin.scss","../../../styles/Configs/Colors.scss","../../../styles/Configs/Fonts.scss"],"names":[],"mappings":"AAKA;ECJE,aAAA;EACA,mBDIqB;ECHrB,mBDG0B;ECF1B,8BDEkC;EAChC,kBAAA;EACA,sBAAA;EACA,yBAAA;AEDJ;AFEI;EGTF,6BAAA;EACA,kBHSyC;EGRzC,gBHQgD;EGPhD,eHOoD;EGNpD,iBHMyD;EACnD,cITY;EJUZ,SAAA;EACA,eAAA;AEIR;AFFI;ECfF,aAAA;EACA,mBDeyB;ECdzB,mBDc8B;ECb9B,8BDasC;AEOxC;AFNQ;EGjBN,yBEDY;EFEZ,kBHiB2C;EGhB3C,gBHgBkD;EGflD,eHesD;EGdtD,iBHc2D;EACjD,cIlBM;EJmBN,eAAA;EACA,sBAAA;EACA,kBAAA;AEYZ;AFVQ;EACI,kBAAA;EACA,WAAA;EACA,YAAA;EACA,kBAAA;AEYZ;AFVQ;EACI,kBAAA;EACA,eAAA;AEYZ;AFVQ;EClCN,aAAA;EACA,mBDkC6B;ECjC7B,mBDiCkC;EChClC,8BDgC0C;EAChC,sBAAA;EACA,kBAAA;AEeZ;AFdY;EACI,WAAA;EACA,YAAA;EACA,kBAAA;AEgBhB;AFdY;EACI,eAAA;AEgBhB","file":"NavbarOnSignIn.module.css"}