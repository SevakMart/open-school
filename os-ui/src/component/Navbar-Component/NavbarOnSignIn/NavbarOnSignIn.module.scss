@import "../../../styles/Mixins/ButtonMixin.scss";
@import "../../../styles/Mixins/FlexMixins.scss";
@import "../../../styles/Mixins/FontMixin.scss";
@import "../../../styles/Configs/Colors.scss";
@import "../../../styles/Configs/Fonts.scss";
.mainContent {
    @include flexStyle(row, center, space-between);
    margin:0 88px;
    padding:28px 0;
    background-color: transparent;
    h2 {
        @include fontStyle($secondary-font,italic,700,20px,24px);
        color:$header-main-color;
        margin:0;
    }
    .navMainContent{
        @include flexStyle(row, center, space-between);
        p{
            @include fontStyle($primary-font,normal,500,16px,24px);
            color:$main-text-color;
            cursor:pointer;
            box-sizing: border-box;
            margin:0 0 0 32px;
        }
        .bookmarkIconStyle {
            margin:0 0 0 25px;
            width:20px;
            height:20px;
            text-align: center;
        }
        .notificationIcon {
            margin:0 0 0 25px;
            cursor:pointer;
        }
        .userInfoContent {
            @include flexStyle(row, center, space-between);
            box-sizing: border-box;
            margin:0 0 0 25px;
            .avatarLogo {
                width:32px;
                height:32px;
                border-radius: 50%;
            }
            .arrowDownIcon {
                cursor:pointer;
            }
        }
    }
}
