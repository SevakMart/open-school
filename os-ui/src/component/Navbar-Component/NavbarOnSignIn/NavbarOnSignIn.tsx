import { useState, useEffect, useRef } from 'react';
import { useTranslation } from 'react-i18next';
import { useNavigate } from 'react-router-dom';
import styles from './NavbarOnSignIn.module.scss';
import ArrowDownIcon from '../../../assets/svg/Arrow-down.svg';
import NotificationIcon from '../../../assets/svg/Notification.svg';
import BookmarkIcon from '../../../icons/Bookmark';
import { SignoutIcon } from '../../../icons/Signout/Signout';
import ProfilePortalContent from '../../ProfilePortalContent/ProfilePortalContent';
import { Portal } from '../../Portal/Portal';

const NavbarOnSignIn = () => {
  const [isOpen, setIsOpen] = useState(false);
  const navigate = useNavigate();
  const arrowRef = useRef(null);
  const { t } = useTranslation();

  const {
    mainContent, navMainContent, userInfoContent,
    bookmarkIconStyle, notificationIcon, avatarLogo, arrowDownIcon,
  } = styles;

  const handleProfilePortal = () => {
    setIsOpen((prevState) => !prevState);
  };

  const popupRef = useRef<HTMLDivElement | null>(null);
  const bodyRef = useRef(document.body);

  const handleClickOutside = (event: MouseEvent) => {
    const target = event.target as Node;
    if (popupRef.current && !popupRef.current.contains(target)) {
      setIsOpen(false);
    }
  };

  useEffect(() => {
    const listener = (event: MouseEvent) => handleClickOutside(event);
    bodyRef.current.addEventListener('mousedown', listener);
    return () => {
      bodyRef.current.removeEventListener('mousedown', listener);
    };
  }, []);

  return (
    <nav className={mainContent}>
      <h2 onClick={() => navigate('/homepage/WhenSignin')}>{t('string.homePage.navBar.logo')}</h2>
      <div className={navMainContent}>
        <p onClick={() => navigate('/exploreLearningPaths')}>{t('string.homePage.navBar.allLearningPaths')}</p>
        <p onClick={() => navigate('/myLearningPath')}>{t('string.homePage.navBar.myLearningPaths')}</p>
        <p onClick={() => navigate('/mentors')}>{t('string.homePage.navBar.mentors')}</p>
        <div className={bookmarkIconStyle}>
          <BookmarkIcon iconSize="20px" />
        </div>
        <img className={notificationIcon} src={NotificationIcon} alt="Notification" />
        <div className={userInfoContent}>
          <img className={avatarLogo} src="https://reactjs.org/logo-og.png" alt="avatar" />
          <img ref={arrowRef} className={arrowDownIcon} style={isOpen ? { transform: 'rotate(180deg)' } : undefined} src={ArrowDownIcon} alt="Arrow down Icon" onClick={handleProfilePortal} />
        </div>
        <div ref={popupRef}>
          <Portal.ProfilePortal isOpen={isOpen}>
            <div ref={popupRef}>
              <ProfilePortalContent icon={<SignoutIcon />} isSignOut>
                {t('string.profilePortal.signOut')}
              </ProfilePortalContent>
            </div>
          </Portal.ProfilePortal>
        </div>
      </div>
    </nav>
  );
};
export default NavbarOnSignIn;
