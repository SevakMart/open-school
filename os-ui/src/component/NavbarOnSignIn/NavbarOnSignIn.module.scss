@import "../../styles/Mixins/ButtonMixin.scss";
@import "../../styles/Mixins/FlexMixins.scss";
@import "../../styles/Mixins/FontMixin.scss";
.mainContent {
    @include flexStyle(row, center, space-between);
    padding-left: 6%;
    padding-right: 6%;
    background-color: transparent;
    h2 {
        @include fontStyle('Helvetica Neue',italic,700,1.25rem,1.5rem);
        color:#222427;
    }
    .navMainContent{
        @include flexStyle(row, center, space-between);
        width:40%;
        p{
            @include fontStyle('Montserrat',normal,500,1rem,1.5rem);
            color:#5E617B;
            cursor:pointer;
            margin-left: 1px;
        }
        .userInfoContent {
            @include flexStyle(row, center, space-between);
            width:10%;
            img {
                width:2rem;
                height:2rem;
                border-radius: 50%;
            }
        }
    }
}
