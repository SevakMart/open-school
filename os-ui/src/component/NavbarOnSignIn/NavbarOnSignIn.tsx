import styles from './NavbarOnSignIn.module.scss';
import {
  APP_LOGO, ALL_LEARNING_PATHS, MY_LEARNING_PATHS, MENTORS,
} from '../../constants/Strings';
import BookmarkIcon from '../../icons/Bookmark';
import Notification from '../../icons/Notification';
import DownArrowIcon from '../../icons/DownArrow';

const NavbarOnSignIn = () => {
  const { mainContent, navMainContent, userInfoContent } = styles;
  return (
    <nav className={mainContent}>
      <h2>{APP_LOGO}</h2>
      <div className={navMainContent}>
        <p>{ALL_LEARNING_PATHS}</p>
        <p>{MY_LEARNING_PATHS}</p>
        <p>{MENTORS}</p>
        <BookmarkIcon iconSize="1.5rem" isBookmarked={false} />
        <Notification />
        <div className={userInfoContent}>
          <img src="https://reactjs.org/logo-og.png" alt="avatar" />
          <DownArrowIcon />
        </div>
      </div>
    </nav>
  );
};
export default NavbarOnSignIn;
