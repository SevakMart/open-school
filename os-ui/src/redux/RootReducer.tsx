import { combineReducers } from 'redux';
import chooseSubcategoryReducer from './Slices/ChoosSubcategorySlice';
import userInfoReducer from './Slices/loginUserSlice';
import allLearningPathFilterParamsReducer from './Slices/AllLearningPathFilterParamsSlice';
import forgotPasswordEmailReducer from './Slices/ForgotPasswordEmailSlice';
import signedUpUserIdReducer from './Slices/SignedUpUserIdSlice';
import suggestedCourseReducer from './Slices/SuggestedCourseSlice';
import userEnrolledCourseByCourseStatusReducer from './Slices/UserEnrolledCourseByCourseStatus';
import allLearningPathCoursesReducer from './Slices/AllLearningPathCourseSlice';
import allMentorsFilterParamsReducer from './Slices/AllMentorsFilterParamsSlice';
import allMentorsSliceReducer from './Slices/AllMentorsSlice';
import savedMentorsReducer from './Slices/SavedMentorsSlice';
import savedCourseReducer from './Slices/SavedLearningPathSlice';
import homepageMentorListReducer from './Slices/HomepageMentorSlice';
import portalStatusReducer from './Slices/PortalOpenStatus';

const rootReducer = combineReducers({
  portalStatus: portalStatusReducer,
  userInfo: userInfoReducer,
  chooseSubcategories: chooseSubcategoryReducer,
  filterParams: allLearningPathFilterParamsReducer,
  forgotPasswordEmail: forgotPasswordEmailReducer,
  signedUpUserId: signedUpUserIdReducer,
  suggestedCourse: suggestedCourseReducer,
  userEnrolledCourseByCourseStatus: userEnrolledCourseByCourseStatusReducer,
  allLearningPathCourses: allLearningPathCoursesReducer,
  allMentorsFilterParams: allMentorsFilterParamsReducer,
  allMentorsList: allMentorsSliceReducer,
  savedMentors: savedMentorsReducer,
  savedCourse: savedCourseReducer,
  homepageMentorList: homepageMentorListReducer,
});
export default rootReducer;
