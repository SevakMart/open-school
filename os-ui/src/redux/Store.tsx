import { configureStore } from '@reduxjs/toolkit';
import {
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import { createStateSyncMiddleware, initMessageListener } from 'redux-state-sync';
import rootReducer from './RootReducer';

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware({
    serializableCheck: {
      ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
    },
  }).concat(
    createStateSyncMiddleware({
      blacklist: ['persist/PERSIST', 'persist/REHYDRATE'],
    }),
  ),
});

initMessageListener(store);

export type RootState = ReturnType<typeof store.getState>
