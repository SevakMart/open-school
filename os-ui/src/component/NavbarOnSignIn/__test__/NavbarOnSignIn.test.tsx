import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import NavbarOnSignIn from '../NavbarOnSignIn';
import { store } from '../../../redux/Store';

test('Create a Snapshot test', () => {
  const { asFragment } = render(<Provider store={store}><NavbarOnSignIn /></Provider>);
  expect(asFragment()).toMatchSnapshot();
});
