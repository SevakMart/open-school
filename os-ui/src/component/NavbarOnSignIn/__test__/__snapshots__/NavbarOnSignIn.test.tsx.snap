// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create a Snapshot test 1`] = `
<DocumentFragment>
  <nav
    class="mainContent"
  >
    <h2>
      Open-School
    </h2>
    <div
      class="navMainContent"
    >
      <p>
        All Learning Paths
      </p>
      <p>
        My Learning Paths
      </p>
      <p>
        Mentors
      </p>
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        style="color: rgb(94, 97, 123); font-size: 1.5rem; cursor: pointer;"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19 21l-7-5-7 5V5a2 2 0 0 1 2-2h10a2 2 0 0 1 2 2z"
        />
      </svg>
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(94, 97, 123); font-size: 1.5rem; cursor: pointer;"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.9 2 2 2zm6-6v-5c0-3.07-1.63-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.64 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2zm-2 1H8v-6c0-2.48 1.51-4.5 4-4.5s4 2.02 4 4.5v6z"
        />
      </svg>
      <div
        class="userInfoContent"
      >
        <img
          alt="avatar"
          src="https://reactjs.org/logo-og.png"
        />
        <i>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(94, 97, 123); cursor: pointer;"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M1.646 4.646a.5.5 0 0 1 .708 0L8 10.293l5.646-5.647a.5.5 0 0 1 .708.708l-6 6a.5.5 0 0 1-.708 0l-6-6a.5.5 0 0 1 0-.708z"
              fill-rule="evenodd"
            />
          </svg>
        </i>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
